Define the `wrap` function for the async example in Best Practices page

davidjb opened the pull request:
This example came from https://strongloop.com/strongblog/async-error-handling-expressjs-es7-promises-generators/ but the definition of `wrap()` wasn't replicated here.  This adds the definition to avoid confusion.

davidjb committed:
Define/explain `wrap` function for async example

This example came from https://strongloop.com/strongblog/async-error-handling-expressjs-es7-promises-generators/ but the definition of `wrap()` wasn't replicated here.  This adds the definition to avoid confusion.

crandmck commented:
Thanks for the PR @davidjb 

Good suggestion, but I'd prefer to add the information outside of the code block.  Something like this right after the code example:
```
The `wrap()` function is a wrapper that catches rejected promises and calls `next()`.  
For details, see [https://strongloop.com/strongblog/async-error-handling-expressjs-es7-promises-generators/#cleaner-code-with-generators](Asynchronous 
Error Handling in Express with Promises, Generators and ES7).
```

davidjb commented:
@crandmck Thanks for the suggestion; updated.  I removed the repeated link to https://strongloop.com/strongblog/async-error-handling-expressjs-es7-promises-generators/ as it would have been there twice in the same text block.
